internal/target/repository.go

```diff
@@ -51,6 +52,12 @@ type Repository struct {
 	permissions []perms.Permission
 }
 
+// getTargetProxyServerCertificateFn can be overridden for testing or extension purposes.
+// By default, it returns nil, nil.
```

Comment:
I'm wondering if we should update this comment to mention that TCP targets do not currently use proxy server certificates, in the case that they will down the line so it's clear this needs to change.